six/modules/c++/six.sicd/tests/test_add_additional_des.cpp

```diff
@@ -0,0 +1,256 @@
+/* =========================================================================
+* This file is part of six.sicd-c++
+* =========================================================================
+*
+* (C) Copyright 2004 - 2016, MDA Information Systems LLC
+*
+* six-c++ is free software; you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; If not,
+* see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <cstdio>
+#include <six/NITFWriteControl.h>
+#include <six/Types.h>
+#include <six/XMLControlFactory.h>
+#include <six/sicd/Utilities.h>
+
+namespace
+{
+void validateArguments(int argc, char** argv)
+{
+    if (argc != 2)
+    {
+        std::string message = "Usage: " + sys::Path::basename(argv[0])
+                + " <XML pathname>";
+        throw except::Exception(Ctxt(message));
+    }
+    if (!str::endsWith(argv[1], ".xml"))
+    {
+        std::string message = "Extension suggests input may not be an XML file. "
+            "Expecting '*.xml'.";
+        throw except::Exception(Ctxt(message));
+    }
+}
+
+std::vector<sys::Int16_T> generateBandData(const six::sicd::ComplexData& data)
```

Comment:
This doesn't matter a ton b/c it's just a test program, but this will potentially result in a copy depending on how smart the compiler is (C++11-compiler will likely optimize this out).